scripts/screeps_tencent_batch_rl_runner.py

```
@@ -2197,10 +2209,12 @@ def verified_remote_runtime_parameter_injection(raw: Any) -> dict[str, Any] | No
     if runtime_injected:
         if status != "injected":
             raise BatchRunError("remote runtimeParameterInjection status must be injected when runtime injection is proven")
-        if not policy_update_eligible:
-            raise BatchRunError("remote runtimeParameterInjection policyUpdateEligible must be true when runtime injection is proven")
         if injected_count <= 0:
             raise BatchRunError("remote runtimeParameterInjection injectedVariantCount must be positive when runtime injection is proven")
+        if policy_update_eligible and runtime_consumed is not True:
+            raise BatchRunError("remote runtimeParameterInjection policyUpdateEligible requires runtimeParameterConsumption=true")
+        if runtime_consumed is True and consumed_count == 0:
+            raise BatchRunError("remote runtimeParameterInjection consumedVariantCount must be positive when consumption is proven")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)